soroban-env-host/src/host.rs

```diff
@@ -541,6 +555,21 @@ impl Host {
         }
     }
 
+    /// Calls the provided function while ensuring that no diagnostic events are
+    /// recorded.
+    /// This is useful for emulating operations only for the sake of budget
+    /// accounting in recording mode.
+    #[cfg(any(test, feature = "recording_mode"))]
+    pub(crate) fn with_suppressed_diagnostic_events<F>(&self, f: F) -> Result<(), HostError>
+    where
+        F: FnOnce() -> Result<(), HostError>,
+    {
+        *self.try_borrow_suppress_diagnostic_events_mut()? = true;
```

Comment:
You might want to observe the previous state and only restore it to previous rather than clobber to false (this will only matter if the function is re-entered, but sometimes that happens with code that has this pattern)

Reply:
Thanks, I'll update this in the followup PR.